### PR TITLE
Port build_directory_md.py to Rust

### PR DESCRIPTION
**Pull request by JADarius**
## Description
This PR ports the `build_directory_md.py` to Rust. I still need guidance with integrating the script with Github Actions.  I have tested the code manually by running `cargo build` in the `build_directory` folder and then running the executable from the root folder (the `Rust` folder). Refactoring suggestions are highly appreciated.

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] I ran bellow commands using the latest version of **rust nightly**.
- [x] I ran `cargo clippy --all -- -D warnings` just before my last commit and fixed any issue that was found.
- [x] I ran `cargo fmt` just before my last commit.
- [x] I ran `cargo test` just before my last commit and all tests passed.
- [x] I checked `COUNTRIBUTING.md` and my code follows its guidelines.

### CODE REVIEW

**Review by siriak — COMMENTED** [review]
You also need to update this script to run Rust script and not the Python one https://github.com/TheAlgorithms/Rust/blob/master/.github/workflows/directory_workflow.yml

**Comment by JADarius** [comment]
I updated the script as requested. I didn't delete the python script. If it's necessary I will do that.

File: .github/workflows/directory_workflow.yml
Comment (50-Course) [inline .github/workflows/directory_workflow.yml:1]: Hi,

outside contributor here, have you check the `v3` of checkout actions?

**Review by 50-Course — COMMENTED** [review]
LGTM, i was thinking is there a reason not to use `v3` of `Checkout` action. Help me understand. Thank you

**Comment by siriak** [comment]
Everything seems to be working. Please delete python scrips and it can be merged

**Comment by JADarius** [comment]
Done. This should fix the first two tasks from https://github.com/TheAlgorithms/Rust/issues/473.

**Review by 50-Course — COMMENTED** [review]
Oh great, thanks for having that off

**Review by siriak — APPROVED** [review]
Looks good, thanks a lot!

**Pull request merged**